src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,223 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String insertCarRequest = "insert into cars (manufacturer_id, model) value (?,?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement insertCarStatement =
+                        connection.prepareStatement(insertCarRequest,
+                                Statement.RETURN_GENERATED_KEYS)) {
+            insertCarStatement.setLong(1,car.getManufacturer().getId());
+            insertCarStatement.setString(2,car.getModel());
+            insertCarStatement.executeUpdate();
+            ResultSet getGeneratedKey = insertCarStatement.getGeneratedKeys();
+            if (getGeneratedKey.next()) {
+                Long id = getGeneratedKey.getObject(1, Long.class);
+                car.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create car: " + car, e);
+        }
+        insertDrivers(car);
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String selectCarWithoutDriversRequest = "select c.id as car_id, "
+                + "m.id as manufacturer_id, "
+                + "m.name as manufacturer_name, "
+                + "m.country as manufacturer_country, "
+                + "c.model as car_model "
+                + "from cars c "
+                + "join manufacturers m on c.manufacturer_id = m.id "
+                + "where c.id = ? and c.is_deleted = false;";
+        Car car = new Car();
```

Comment:
```suggestion
        Car car = null;
```

we need to return empty Optional in case there is no car with such id